axum-macros/src/debug_handler.rs

```diff
@@ -284,6 +286,169 @@ fn check_inputs_impls_from_request(item_fn: &ItemFn, state_ty: Type) -> TokenStr
         .collect::<TokenStream>()
 }
 
+///If the output is a tuple with 2 or more elements,
+/// it checks with the following pattern,
+/// first element => StatusCode || Parts || IntoResponseParts
+///last element => IntoResponse
+///other elements => IntoResponseParts
+///the max numbers of IntoResponseParts(16)
+fn check_output_tuples(item_fn: &ItemFn) -> Option<TokenStream> {
+    //Extract tuple types
+    let elements = match &item_fn.sig.output {
+        ReturnType::Type(_, ty) => match &**ty {
+            Type::Tuple(tuple) => &tuple.elems,
+            _ => return None,
+        },
+        _ => return None,
+    };
+
+    if elements.len() < 2 {
+        return None;
+    }
```

Comment:
This this special case? What if the return type is `(String,)`, its a bit exotic but if it doesn't require much special casing then it would be cool to support.

Reply:
My thoughts were that the origin error message for this case is already specific enough.
ex: (String,) does not implement IntoResponse.
But it'd be a great idea to make a early return and the error more specific, so I added the check.

Reply:
Wait I just noticed that (String,)  does not implement IntoResponse.
Do you want a check for String if it implements IntoResponse and suggest removing the parentheses  ?

Reply:
It does on main https://github.com/tokio-rs/axum/pull/2143

Reply:
Generally the length of the tuple shouldn’t matter. (String,) should work with the same code path as any other tuple I think 🤔

Reply:
Why do you need to special case 1 element tuples? Shouldn't the same code work for 1/2/3/... length tuples? I mean you're just doing loops and maps and stuff.

Reply:
I wasn't familiar with the WithPosition Api and didn't notice that when there's only one element it falls to Position::Only instead of Position::First, concerning it might cause unknown behaviors.
After discovering this , I deleted the length check . The code should be more readable rn , thanks for the comment!